### PR TITLE
feat(Core/AI): Implement delay option to the Talk() function

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:
-  Implement a delay option to the talk function, allowing us to create simple dialogues without the need of using the EventScheduler or other more sophisticated event scheduling
-  Example:
- 
image
boss_skarloc.cpp

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
Doxygen comment? 😏

**Comment by Nyeriah** [comment]
Done and also added example code

**Review by Kitzunu — APPROVED** [review]

File: src/server/game/AI/CreatureAI.cpp [hunk 1]
```diff
@@ -40,9 +40,26 @@ void CreatureAI::OnCharmed(bool /*apply*/)
...
+ * @brief Causes the creature to talk/say the text assigned to their entry in the `creature_text` database table.
+ *
+ * @param uint8 id Text ID from `creature_text`.
+ * @param WorldObject target The target of the speech, in case it has elements such as $n, where the target's name will be referrenced.
+ * @param Milliseconds delay Delay until the creature says the text line. Creatures will talk immediately by default.
+ */
+void CreatureAI::Talk(uint8 id, WorldObject const* target /*= nullptr*/, Milliseconds delay /*= 0s*/)
 {
-    sCreatureTextMgr->SendChat(me, id, target);
+    if (delay > Seconds::zero())
```
Comment (Winfidonarleyan) [inline src/server/game/AI/CreatureAI.cpp:52]: can use `0s` or `0ms`

Comment (Nyeriah) [reply src/server/game/AI/CreatureAI.cpp:52]: Other functions in the core use Secods::zero() (as ported from Trinitycore)

Comment (Winfidonarleyan) [reply src/server/game/AI/CreatureAI.cpp:52]: okay

**Pull request merged**